pyccel/stdlib/gFTL_functions/Map_extensions.inc

```diff
@@ -0,0 +1,48 @@
+#include <map/header.inc>
+
+contains
+
+#define __IDENTITY(x) x
+#define __guard __map_guard
+#include "parameters/Key/copy_map_Key_to_internal_Key.inc"
+#include "parameters/T/copy_map_T_to_internal_T.inc"
+#include "parameters/Key/define_derived_macros.inc"
+#include "parameters/T/define_derived_macros.inc"
+
+  function __IDENTITY(Map)_pop(my_map, key) result(value)
+    class(Map), intent(inout) :: my_map
+    __Key_declare_dummy__, intent(in) :: key
+
+    __T_declare_dummy__ :: value
+
+    type (__map_iterator) :: iter
+    type (__map_iterator) :: new_iter
+
+    iter = my_map % find( key )
+    value = iter % second()
+    new_iter = my_map % erase( iter )
```

Comment:
If `new_iter` is just a local iterator, why do we need it?

Reply:
`new_iter` is not the result, though. The result of this function is `value`.

Reply:
I am talking about the function `my_map % erase()`. Fortran doesn't let us do
```fortran
my_map % erase( iter )
```

And I'm not certain how well it works if I do:
```fortran
iter = my_map % erase( iter )
```
If the latter should work without any issues due to `iter` being passed as argument and used to save the result then I agree that we should use this

Reply:
I see. We would need a subroutine version of `erase`...

Reply:
To remove the result entirely, yes. But this function is provided by gFTL and I don't think it's worth creating a `my_dict_type_erase_subroutine` method just for this

Reply:
In my previous comment I meant that gFTL should provide one 😉

Reply:
`new_iter` removed: 044131b5c4ab62d563df3225408582778ce6d811